### PR TITLE
Semester Management

### PR DESCRIPTION
**Pull request by victor-paumier**
- Ask BCs for month of semester instead of current semester
- Change semester automatically, 4 weeks before the end of the current semester
- Improve semester selector on registration page by specify months and year
- Make arrival date optional

### CODE REVIEW

File: assets/js/section.js [hunk 1]
```diff
@@ -1,52 +1,37 @@
...
+    tools.initDatatable(table, {
+        order: [[0, 'asc']],
+        processing: loader.start(),
+        serverSide: true,
+        ajax: Routing.generate('section_list'),
+        columns,
+        columnDefs: [
+            { orderable: false, targets: unsortable },
+        ],
+    }, 'BuddySystem - List Sections');
```
Comment (JeremieSamson) [inline assets/js/section.js]: Sections list ?

File: src/Command/SemesterSwitcherCommand.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        $this->mailManager = $mailManager;
+        $this->semesterHelper = $semesterHelper;
+        parent::__construct($name);
+    }
+
+    public function configure()
+    {
+        $this
+            ->setName('cron:semester:switch')
+            ->setDescription('If the date of end of semester is reached, switch to the next one. This cron will be execute every monday morning (4am)')
```
Comment (JeremieSamson) [inline src/Command/SemesterSwitcherCommand.php]:
```suggestion
            ->setDescription('If the date of end of semester is reached, switch to the next one. This cron will be executed every monday morning at 4am')
```

File: src/Entity/Section.php [hunk 3]
```diff
@@ -119,13 +126,17 @@ class Section
     private $motivations;
 
     /**
+     * @Assert\Range(min=1)
```
Comment (JeremieSamson) [inline src/Entity/Section.php]: nbMentor & nbMentees should be unsigned and maybe Range is not the good Assert here, maybe `GreaterThan` ?

File: src/Entity/Section.php [hunk 4]
```diff
@@ -119,13 +126,17 @@ class Section
...
+     *
      * @var int
      *
      * @ORM\Column(name="nbMentors", type="integer", nullable=true)
      */
     private $nbMentors;
 
     /**
+     * @Assert\Range(min=1)
```
Comment (JeremieSamson) [inline src/Entity/Section.php]: same here ?

File: src/Entity/User.php [hunk 5]
```diff
@@ -544,14 +557,14 @@ public function getComment(): ?string
         return $this->comment;
     }
 
-    public function setArchive($archive): void
+    public function setArchived($archived): void
```
Comment (JeremieSamson) [inline src/Entity/User.php]: typehint

File: assets/js/section.js (hunk 1 above)
Comment (victor-paumier) [reply assets/js/section.js]: Don't know but should be the same everywhere. Currently with have "List Users" & "Buddies list", seems incoherent

File: src/Command/SemesterSwitcherCommand.php (hunk 2 above)
Comment (victor-paumier) [reply src/Command/SemesterSwitcherCommand.php]: done

File: src/Entity/Section.php (hunk 3 above)
Comment (victor-paumier) [reply src/Entity/Section.php]: done

File: src/Entity/Section.php (hunk 4 above)
Comment (victor-paumier) [reply src/Entity/Section.php]: done

File: src/Entity/User.php (hunk 5 above)
Comment (victor-paumier) [reply src/Entity/User.php]: done

File: assets/js/section.js (hunk 1 above)
Comment (JeremieSamson) [reply assets/js/section.js]: We should have `Users list ` and `Buddie list` to be coherent then

Comment (victor-paumier) [reply assets/js/section.js]: done

**Pull request merged**